lib/node_modules/@stdlib/array/complex64/test/test.slice.js

```diff
@@ -0,0 +1,262 @@
+var Float32Array = require( '@stdlib/array/float32' );
+var Complex64Array = require( './../lib' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof Complex64Array, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'attached to the prototype of the main export is a `slice` method', function test( t ) {
+	t.strictEqual( hasOwnProp( Complex64Array.prototype, 'slice' ), true, 'has property' );
+	t.strictEqual( isFunction( Complex64Array.prototype.slice ), true, 'has method' );
+	t.end();
+});
+
+tape( 'the method throws an error if invoked with a `this` context which is not a complex number array instance', function test( t ) {
+	var values;
+	var arr;
+	var i;
+
+	arr = new Complex64Array( 5 );
+
+	values = [
+		'5',
+		5,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[],
+		function noop() {}
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.slice.call( value );
+		};
+	}
+});
+
+tape( 'the method throws an error if provided a first argument which is not an integer', function test( t ) {
+	var values;
+	var arr;
+	var i;
+
+	arr = new Complex64Array( 10 );
+
+	values = [
+		'5',
+		3.14,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[]
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.slice( value );
+		};
+	}
+});
+
+tape( 'the method throws an error if provided a second argument which is not an integer', function test( t ) {
+	var values;
+	var arr;
+	var i;
+
+	arr = new Complex64Array( 10 );
+
+	values = [
+		'5',
+		3.14,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[]
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.slice( 0, value );
+		};
+	}
+});
+
+tape( 'the method returns empty typed array if operating on an empty complex number array', function test( t ) {
+	var arr;
+	var out;
+
+	arr = new Complex64Array();
+	out = arr.slice();
+
+	t.strictEqual( out.length, 0, 'returns expected value' );
+	t.end();
+});
+
+tape( 'if called without arguments, the method returns a typed array containing the same elements as the original array', function test( t ) {
+	var expected;
+	var actual;
+	var arr;
+
+	arr = new Complex64Array( [ 1.0, -1.0, 2.0, -2.0, 3.0, -3.0 ] );
+	expected = new Float32Array( [ 1.0, -1.0, 2.0, -2.0, 3.0, -3.0 ] );
+	actual = arr.slice();
+
+	t.strictEqual( instanceOf( actual, Complex64Array ), true, 'returns expected value' );
```

Comment:
```suggestion
	t.strictEqual( instanceOf( actual, Complex64Array ), true, 'returns expected value' );
	t.notEqual( actual, arr, 'returns a new instance' );
```

We should explicitly test that a new instance is returned.